### PR TITLE
fix: pass element in listener field injections

### PR DESCRIPTION
**Pull request by Skaiir**
Related to https://github.com/camunda/camunda-modeler/issues/2989

### CODE REVIEW

File: test/spec/provider/camunda-platform/FieldInjectionProps.spec.js [hunk 1]
```diff
@@ -64,6 +65,38 @@ describe('provider/camunda-platform - FieldInjectionProps', function() {
     debounceInput: false
   }));
 
+  describe('within ExecutionListener', function() {
+
+    it('should update field value', inject(async function(elementRegistry, selection) {
+
+      // given
+
```
Comment (barmac) [inline test/spec/provider/camunda-platform/FieldInjectionProps.spec.js:73]:
```suggestion
```

File: test/spec/provider/camunda-platform/FieldInjectionProps.spec.js [hunk 2]
```diff
@@ -64,6 +65,38 @@ describe('provider/camunda-platform - FieldInjectionProps', function() {
...
+      const serviceTask = elementRegistry.get('ServiceTask_ExecutionListener');
+
+      await act(() => {
+        selection.select(serviceTask);
+      });
+
+      // when
+
```
Comment (barmac) [inline test/spec/provider/camunda-platform/FieldInjectionProps.spec.js:81]:
```suggestion
```

File: test/spec/provider/camunda-platform/FieldInjectionProps.spec.js [hunk 3]
```diff
@@ -64,6 +65,38 @@ describe('provider/camunda-platform - FieldInjectionProps', function() {
...
+      const executionListenerGroup = domQuery('[data-group-id="group-CamundaPlatform__ExecutionListener"]', container);
+      const valueInputField = domQuery('input[name*="value"]', executionListenerGroup);
+
+      changeInput(valueInputField, 'newValue');
+
+      // then
+
```
Comment (barmac) [inline test/spec/provider/camunda-platform/FieldInjectionProps.spec.js:88]:
```suggestion
```

File: test/spec/provider/camunda-platform/FieldInjectionProps.spec.js [hunk 4]
```diff
@@ -64,6 +65,38 @@ describe('provider/camunda-platform - FieldInjectionProps', function() {
...
+      const extensionElements = serviceTask.businessObject.extensionElements.values;
+      expect(extensionElements.length).to.equal(1);
```
Comment (barmac) [inline test/spec/provider/camunda-platform/FieldInjectionProps.spec.js:90]:
```suggestion
      expect(extensionElements).to.have.lengthOf(1);
```

File: test/spec/provider/camunda-platform/FieldInjectionProps.spec.js [hunk 5]
```diff
@@ -64,6 +65,38 @@ describe('provider/camunda-platform - FieldInjectionProps', function() {
...
+
+      const executionListener = extensionElements[0];
+      expect(executionListener.fields.length).to.equal(1);
```
Comment (barmac) [inline test/spec/provider/camunda-platform/FieldInjectionProps.spec.js:93]:
```suggestion
      expect(executionListener.fields).to.have.lengthOf(1);
```

File: test/spec/provider/camunda-platform/FieldInjectionProps.spec.js [hunk 6]
```diff
@@ -64,6 +65,38 @@ describe('provider/camunda-platform - FieldInjectionProps', function() {
     debounceInput: false
   }));
 
+  describe('within ExecutionListener', function() {
+
+    it('should update field value', inject(async function(elementRegistry, selection) {
```
Comment (barmac) [inline test/spec/provider/camunda-platform/FieldInjectionProps.spec.js:70]: Can we have a test case for the name, too?

File: src/provider/camunda-platform/properties/ListenerProps.js [hunk 7]
```diff
@@ -414,6 +414,7 @@ function Field(props) {
   return (
     <CollapsibleEntry
       id={ fieldId }
+      element={ element }
```
Comment (barmac) [inline src/provider/camunda-platform/properties/ListenerProps.js:417]: So indeed we could have detected this if there'd been some mechanism for required props... 😓

**Review by barmac — CHANGES_REQUESTED** [review]
Let's add the test case for name. Feel free to also merge a couple of suggestions I left (optional).

File: src/provider/camunda-platform/properties/ListenerProps.js (hunk 7 above)
Comment (Skaiir) [reply src/provider/camunda-platform/properties/ListenerProps.js:417]: Well in any case element isn't technically required here, if your implementation of adders and setters doesn't make use of it. CollapsibleEntry is properties-panel core, from my understanding it should be completely unaware of bpmn-js concerns.

**Review by barmac — APPROVED** [review]

**Pull request merged**